### PR TITLE
Phdo 756 get upload stats

### PR DESCRIPTION
**Pull request by cyber-decker**
# PHDO-756: Playwright Test Implementation for getUploadStats

This branch adds comprehensive Playwright end-to-end tests for the new `getUploadStats` 

### getUploadStats tests `getUploadStats.spec.ts`
  - Empty results for non-existing data streams
  - Error handling for invalid parameters
  - Multiple completed upload reports
  - Unique upload counting
  - Pending upload detection
  - Date range filtering (start date, end date, interval)
  - Bad metadata counting
  - Undelivered upload identification
  - Duplicate filename handling

### 🔧 Updated test helpers
- **`dataGenerator.ts`**: new helper functions for creating test data
- **`reportHelper.ts`**: validation and cleanup utilities

## Summary
This test implementation ensures the `getUploadStats` functionality is thoroughly validated across all supported environments, providing confidence in the feature's reliability and consistency. The comprehensive test suite covers both happy path scenarios and edge cases, with automated snapshot validation to catch regressions.

### CODE REVIEW

File: test/playwright/fixtures/dataGenerator.ts [hunk 1]
```diff
@@ -131,22 +153,47 @@ export function createUploadReportCompleted(report?: UploadReport): UploadReport
...
+export function createStageInfo(date: Date = new Date(), overrides?: Partial<StageInfo>) {
+    const defaultStageInfo = createStageInfoStarted(date)
+    return { ...defaultStageInfo, ...overrides }
 }
 
-export function createStageInfoStarted(date: Date = new Date()) {
-    const stage_info =  {
+export function createStageInfoStarted(date: Date = new Date(), overrides?: Partial<StageInfo>) {
+    const defaultStageInfo = {
         service: "UPLOAD API",
```
Comment (mkrystof) [inline test/playwright/fixtures/dataGenerator.ts]: Would it make sense to introduce some constants for the service and action? Just thinking from a maintenance standpoint it would be nice to have these all in one place so if it changes we don't have to search through and change all the places in the code "UPLOAD API", for example, occurs.

File: test/playwright/tests/getUploadStats.spec.ts [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        const result = await gql.getUploadStats({
+            dataStreamId: "nonexisting",
+            dataStreamRoute: "nonexisting"
+        }, { failOnEmptyData: false }) as unknown as GraphQLErrorResponse;
+        expect(result.errors[0].message).toContain("Parsing of the input failed");
+        expect(result.errors[0].extensions.classification).toBe("ParsingFailureException");
+        expect(result.errors[0].path).toStrictEqual(["getUploadStats"]);
+    });
+
+    test('returns stats for multiple completed upload reports', async ({ gql, reportHelper }) => {
```
Comment (mkrystof) [inline test/playwright/tests/getUploadStats.spec.ts:23]: Would be nice to see another test similar to this one where instead of using the "create" action we "replace" three reports and makes sure that the total uploads only increases by one.

**Review by mkrystof — CHANGES_REQUESTED** [review]
Looking great! Just a couple suggestions.

File: test/playwright/fixtures/dataGenerator.ts (hunk 1 above)
Comment (cyber-decker) [reply test/playwright/fixtures/dataGenerator.ts]: This is an easy change.  I can extract this to a better place to have the "default" so they all reference the same default block and make the parts that are variable a little more variable.  There's also the capability to override this too, which is nice if the default is ever not appropriate.  

The defaultStageInfo now exists as it's own block in dataGenerator and that is the only place it is referenced.  If we need to update it for everything, that's where to do it.  All other functions now use this piece and add on whatever else is necessary.  An override can also be done like this: `dataGenerator.createStageInfoStarted(new Date(), { service: "Test Service"})`

**Review by mkrystof — APPROVED** [review]
Awesome work as usual, thank you!
